app/src/openxr/cpp/OpenXRInputSource.cpp

```diff
@@ -800,6 +800,18 @@ XrResult OpenXRInputSource::UpdateInteractionProfile(ControllerDelegate& delegat
         delegate.SetHapticCount(mIndex, numHaptics);
     }
 
+    // On emulated profiles we need to set the button count here because it
+    // may never be set during Update() (e.g, when hand tracking is active).
```

Comment:
I think it's more consistent to move this to `EmulateControllerFromHand` which is kind of the replacement for `Update` for hand tracking

Reply:
Hmm, the problem is that `EmulateControllerFromHand` runs on every update (frame), but in the case of hand tracking this is only necessary once, because it never changes like in the case of actual controllers.

So adding this count loop to every Update() seems a bad idea from a perf point of view. WDYT?

Reply:
I have left the button count code in `UpdateInteractionProfile()` based on the reasoning above. Let me know if you think I should move it to `EmulateControllerFromHand()` anyway.